### ISSUE TITLE
no-concatenated-classes detects false positive

### ISSUE THREAD

**Issue by FabienDehopre** [schoero/eslint-plugin-readable-tailwind#392]
### Description

when using ternary operator in template, the `no-concatenated-classes` rule triggers an error even if the expression is valid (based on the official documentation).

### Flavor

Angular

### Code Input

```html
<div [class]="'flex w-full min-w-0 ' + (menuOpen() ? 'gap-1' : 'flex-col gap-0')">
...
</div>
```

### Expected Behavior

The rule should not mark this construct as bad because it's valid based on https://tailwindcss.com/docs/detecting-classes-in-source-files#dynamic-class-names

### Actual Behavior

The rule marks this construct as invalid.

### ESLint Log

```shell
62:72  error  Concatenated classes may be purged by Tailwind CSS. Avoid dynamic class construction  better-tailwindcss/no-concatenated-classes
```

### ESLint Config

I'm using a custom config package that I've build for all my projects. You can find the tailwindcss config here: https://github.com/FabienDehopre/eslint-config/blob/main/src/configs/tailwindcss.ts

### Versions

- ESLint: 10.7.0
- Parser: @angular-eslint/template-parser@22.1.0
- Plugin: 4.7.0
- Node: 24.18.0
